src/etcd/client.py

```diff
@@ -180,7 +181,19 @@ def uri(protocol, host, port):
         elif password:
             _log.warning('Password provided without username, both are required for authentication')
 
-        self.http = urllib3.PoolManager(num_pools=10, **kw)
+        # If a proxy environment variable is set for protocol,
+        # then respect it
+        if protocol == 'https':
+            _proxy_vars = ['https_proxy', 'HTTPS_PROXY']
+        else:
+            _proxy_vars = ['http_proxy', 'HTTP_PROXY']
+
+        for _proxy_var in _proxy_vars:
+            if _proxy_var in os.environ:
+                self.http = urllib3.ProxyManager(os.environ[_proxy_var], num_pools=10, **kw)
+                break
```

Comment:
I think it would be nice to have a ``_log`` statement here. Either ``debug`` or ``info`` depending on the verbosity of information. That way operators can be sure they are going through the proxy via log trail.